src/wallet/coin_selection.rs

```diff
@@ -1339,4 +1495,140 @@ mod test {
         assert!(result.selected_amount() > target_amount);
         assert_eq!(result.fee_amount, (50 + result.selected.len() * 68) as u64);
     }
+
+    #[test]
+    fn test_calculate_waste_with_change() {
+        let fee_rate = FeeRate::from_sat_per_vb(11.0);
+        let selected = generate_utxos_of_values(vec![100_000_000, 200_000_000]);
+        let amount_needed = 200_000_000;
+
+        let utxos: Vec<OutputGroup<'_>> = selected
+            .iter()
+            .map(|u| OutputGroup::new(u, fee_rate))
+            .collect();
+
+        let size_of_change = 31;
+        let cost_of_change: u64 = (size_of_change as f32 * fee_rate.as_sat_vb()) as u64;
+
+        let utxo_fee_diff: i64 = utxos.iter().fold(0, |acc, utxo| {
+            let fee_rate: i64 = utxo.fee as i64;
```

Comment:
I'm confused at this variable name. Shouldn't this be a `fee` rather than a `fee_rate`? (Also in the following test cases.)

Reply:
You are right.